http/client/src/main/java/org/eclipse/rdf4j/http/client/BackgroundResultExecutor.java

```diff
@@ -0,0 +1,68 @@
+package org.eclipse.rdf4j.http.client;
+
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+
+import org.eclipse.rdf4j.query.GraphQueryResult;
+import org.eclipse.rdf4j.query.QueryResult;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.query.impl.BackgroundGraphResult;
+import org.eclipse.rdf4j.query.resultio.TupleQueryResultParser;
+import org.eclipse.rdf4j.query.resultio.helpers.BackgroundTupleResult;
+import org.eclipse.rdf4j.rio.RDFParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BackgroundResultExecutor implements AutoCloseable {
+
+	private final Logger logger = LoggerFactory.getLogger(BackgroundResultExecutor.class);
+
+	private final ExecutorService executor;
+
+	private final HashSet<QueryResult<?>> executing = new HashSet<>();
```

Comment:
This Set may need to be threadsafe, or synchronized access used. Concurrent calls to add remove are possible currently and HashSet on its own cannot handle that consistently.